crates/wasi-provider/src/lib.rs

```diff
@@ -18,22 +18,35 @@ use handle::PodHandle;
 use wasi_runtime::WasiRuntime;
 
 const TARGET_WASM32_WASI: &str = "wasm32-wasi";
+const LOG_DIR_NAME: &str = "wasi-logs";
 
 /// WasiProvider provides a Kubelet runtime implementation that executes WASM
 /// binaries conforming to the WASI spec
 #[derive(Clone)]
 pub struct WasiProvider {
     handles: Arc<RwLock<HashMap<String, PodHandle<File>>>>,
     store: FileModuleStore,
+    log_path: PathBuf,
 }
 
 impl WasiProvider {
-    pub fn new() -> Self {
-        let store = FileModuleStore::new(&dirs::home_dir().expect("Cannot get home directory"));
-        Self {
+    /// Returns a new WASI provider configured to use the proper data directory
+    /// (including creating it if necessary)
+    pub fn new<P: AsRef<Path>>(data_dir: P) -> anyhow::Result<Self> {
+        // Make sure we have a log dir and containers dir created
+        let data_path = data_dir.as_ref().to_path_buf();
+        // This is temporary as we should probably be passing in a ModuleStore
+        // as a parameter or the oci client as a whole
+        // NOTE: We do not have to create the dir here as the FileModuleStore already does this
+        let container_path = data_path.join("containers");
```

Comment:
Can we put `"containers"` in a const?

Reply:
I didn't on purpose because this is temporary. If you feel strongly about it, I'll add it in

Reply:
Nope. Shall we create an issue?

Reply:
Yeah, I'll create one as an output of the discussion we have today offline